ffs-shared/rule/src/commonMain/kotlin/rule/IpParser.kt

```diff
@@ -0,0 +1,55 @@
+package doist.ffs.rule
+
+import kotlin.experimental.and
+import kotlin.experimental.inv
+import kotlin.experimental.or
+import kotlin.math.pow
+
+fun calculateIpv4Value(ipSegments: List<Short>): Long {
+    var result = 0L
+    ipSegments.forEachIndexed { i, value ->
+        result += value * (256f.pow(3 - i)).toLong()
+    }
+
+    return result
+}
+
+fun ipv4ToNumeric(ipAddress: String): Long {
+    val splittedIpv4 = ipAddress.split('.').map { it.toShort() }
+    if (splittedIpv4.size != 4 || splittedIpv4.any { it < 0 || it > 255 }) {
+        throw IllegalArgumentException("invalid IPv4 format")
+    }
+
+    return calculateIpv4Value(splittedIpv4)
+}
+
+fun cidrv4ToRange(cidr: String): LongRange {
```

Comment:
Sure we can do it inline inside Data Class, but we still need to have the `calculateIpv4Value` function, because we're using `List<Short>` and `reduce` only supports calculation in the same data type

Unless we're doing `.map` to `Long` beforehand, which will add more iteration to the calculation but will have a cleaner looking code

What do you think?

Reply:
Calculation function will look like this
```
private fun calculateIpv4Value(octets: List<UByte>): Long {
    return octets.map { it.toLong() }.reduce { result, octet -> (result shl 8) + octet }
}
```

Reply:
> Unless we're doing .map to Long beforehand, which will add more iteration to the calculation but will have a cleaner looking code

Agree we shouldn't, it's wasteful. But I'm puzzled by `List<Short>`. Why can't we use `List<UByte>` everywhere?